src/actinia_statistic_plugin/vector_sampling.py

```diff
@@ -0,0 +1,233 @@
+
+    decorators = [log_api_call, auth.login_required]
+
+    @swagger.doc(deepcopy(SCHEMA_DOC))
+    def post(self, location_name, mapset_name, vector_name):
+        """Perform vector map sampling on a vector map layer based on input points synchronously"""
+        check = self._execute(location_name, mapset_name, vector_name)
+        if check is not None:
+            http_code, response_model = self.wait_until_finish()
+        else:
+            http_code, response_model = pickle.loads(self.response_data)
+        return make_response(jsonify(response_model), http_code)
+
+
+def start_job(*args):
+    processing = AsyncEphemeralVectorSampling(*args)
+    processing.run()
+
+
+class AsyncEphemeralVectorSampling(EphemeralProcessing):
+    """Sample a vector map at vector points"""
+
+    def __init__(self, *args):
+        EphemeralProcessing.__init__(self, *args)
+        self.response_model_class = VectorSamplingResponseModel
+
+    def _execute(self):
+
+        self._setup()
+
+        # Points are stored in self.request_data
+        vector_name = self.map_name
+        points = self.request_data["points"]
+
+        if not points or len(points) == 0:
+            raise AsyncProcessError("Empty coordinate list")
+
+        coordinates_string = ""
+        for tuple in points:
+            if len(tuple) != 3:
+                raise AsyncProcessError("Wrong number of coordinate entries")
+
+            id, x, y = tuple
+            coordinates_string += "%s,%s," % (x, y)
+
+        # g.region probably not needed
+        pc = {
+            "list": [
+                {
+                    "id": "g_region",
+                    "module": "g.region",
+                    "inputs": [
+                        {
+                            "param": "vector",
+                            "value": "%s@%s" % (vector_name, self.mapset_name)
+                        }
+                    ],
+                    "flags": "p",
+                },
+                {
+                    "id": "v_what",
+                    "module": "v.what",
+                    "inputs": [
+                        {
+                            "param": "map",
+                            "value": "%s@%s" % (vector_name, self.mapset_name)
+                        },
+                        {
+                            "param": "coordinates",
+                            "value": coordinates_string[:-1]
+                        }
+                    ],
+                    "stdout":
+                        {
+                            "id": "info",
+                            "format": "list",
+                            "delimiter": "|"
+                        },
+                    "flags": "ag",
+                },
+            ],
+            "version": "1",
+        }
+
+        self.request_data = pc
+
+        # Run the process chain
+        EphemeralProcessing._execute(self, skip_permission_check=True)
```

Comment:
It is save to skip the permission check because this is handled elsewhere, correct? We are calling it internally and therefore no check needed, right?

Reply:
I think it is checked before, but I am not 100 percent sure

Reply:
I suggest to move this to a separate issue/PR as it concerns all endpoints.

Reply:
Added as new issue #9